src/bsoncxx/test/bson_value.cpp

```diff
@@ -54,6 +54,28 @@ TEST_CASE("types::bson_value::value", "[bsoncxx::types::bson_value::value]") {
         SECTION("bool") {
             auto test_doc = bson_value::make_value(types::b_bool{true});
             value_construction_test(test_doc.view());
+
+            SECTION("with explicit BSON type") {
+                bson_value::value copy_initialization = types::b_bool{true};
+                REQUIRE(copy_initialization == test_doc);
+
+                bson_value::value direct_initialization(types::b_bool{true});
+                REQUIRE(direct_initialization == test_doc);
+
+                bson_value::value _static_cast = (bson_value::value)types::b_bool{true};
+                REQUIRE(_static_cast == test_doc);
+            }
+
+            SECTION("with implicit BSON type") {
```

Comment:
These implicit tests won't exist for types like `symbol`, `regex`, etc.

Reply:
Acknowledged.  I would suggest writing convenience methods to do the `implicit_construction_test()` and `explicit_construction_test()`, similar to the `value_construction_test()` method in this file, so we can run the same tests easily for every type.  Writing that method will involve some (fun!) template meta-programming, I'm happy to give further suggestions on that or pair program it with you some time.

Reply:
Done